db/migrate/20150610131245_change_start_and_end_time_fields_on_shift_to_datetime.rb

```diff
@@ -0,0 +1,15 @@
+class ChangeStartAndEndTimeFieldsOnShiftToDatetime < ActiveRecord::Migration
+  def up
+    remove_column :shifts, :starting_time, :time
+    remove_column :shifts, :ending_time, :time
+    add_column :shifts, :starting_time, :datetime
+    add_column :shifts, :ending_time, :datetime
+  end
+
+  def down
+    remove_column :shifts, :starting_tame, :datetime
+    remove_column :shifts, :ending_time, :datetime
+    add_column :shifts, :starting_time, :time
+    add_column :shifts, :ending_time, :time
+  end
+end
```

Comment:
But surely the date component will come from the RotaSlot? Whatever date part is pulled in by AR will be thrown away anyway.

I think we'll need to have a method on RotaSlot called `starting_time` or something:

```ruby
def starting_time
  DateTime.new(time_part_from_shift, date_part_from_slot)
end
```

and we compare the DateTime passed in via the API with that?